src/components/navigation/section-document-or-browser.tsx

```diff
@@ -0,0 +1,259 @@
+const kTabSectionBorderWidth = 2;
+
+interface IProps {
+  tabSpec: NavTabSpec;
+  reset?: () => void;
+  selectedDocument?: string;
+  selectedSection?: ENavTabSectionType;
+  onSelectNewDocument?: (type: string) => void;
+  onSelectDocument?: (document: DocumentModelType) => void;
+  onTabClick?: (title: string, type: string) => void;
+  isChatOpen?: boolean;
+}
+
+export interface ISubTabSpec {
+  label: string;
+  sections: NavTabSectionSpec[];
+}
+
+export const SectionDocumentOrBrowser: React.FC<IProps> = ({ tabSpec, reset, selectedDocument,
+    onSelectNewDocument, onSelectDocument, onTabClick }) => {
+  const [referenceDocument, setReferenceDocument] = useState<DocumentModelType>();
+  const [ tabIndex, setTabIndex ] = useState(0);
+  const appConfigStore = useAppConfig();
+  const problemStore = useProblemStore();
+  const context = useUserContext();
+  const queryClient = useQueryClient();
+  const ui = useUIStore();
+  const user = useUserStore();
+  const classStore = useClassStore();
+  const navTabSpec = appConfigStore.navTabs.getNavTabSpec(tabSpec.tab);
+  const subTabs: ISubTabSpec[] = [];
+  // combine sections with matching titles into a single tab with sub-sections
+  tabSpec.sections?.forEach(section => {
+    const found = subTabs.findIndex(tab => tab.label === section.title);
+    if (found >= 0) {
+      subTabs[found].sections.push(section);
+    }
+    else {
+      subTabs.push({ label: section.title, sections: [section] });
+    }
+  });
+  const hasSubTabs = subTabs.length > 1;
+  const vh = window.innerHeight;
+  const headerOffset = hasSubTabs
+                        ? kHeaderHeight + (2 * (kWorkspaceContentMargin + kNavTabHeight + kTabSectionBorderWidth))
+                        : kHeaderHeight + kNavTabHeight + (2 * (kWorkspaceContentMargin + kTabSectionBorderWidth));
+  const documentsPanelHeight = vh - headerOffset;
+  const documentsPanelStyle = { height: documentsPanelHeight };
+  const sectionClass = referenceDocument?.type === "learningLog" ? "learning-log" : "";
+  useEffect(()=>{
+    const selectedSection = tabSpec.tab === "supports" ? ENavTabSectionType.kTeacherSupports : undefined;
+    if (selectedSection) {
+      const selectedIndex = tabSpec.sections?.findIndex(spec => spec.type === selectedSection);
+      if (selectedIndex != null) {
+        setTabIndex(selectedIndex);
+      }
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  },[]);
+  useEffect(()=>{
+    if (reset) {
+      console.log("in reset:", tabSpec.label, reset);
+      // setTimeout to avoid infinite render issues
+      reset();
+      setTimeout(() => handleTabClick(tabSpec.label));
+    }
+  }, [reset]);
+
+
+  const handleTabClick = (title: string, type?: string) => {
+    setReferenceDocument(undefined);
+    ui.updateFocusDocument();
+    ui.setSelectedTile();
+    Logger.log(LogEventName.SHOW_TAB_SECTION, {
+      tab_section_name: title,
+      tab_section_type: type
+    });
+  };
```

Comment:
Probably want to `useCallback()` here for purposes of `useEffect` dependency above.